cmd/timescale-prometheus/main.go

```diff
@@ -208,18 +208,44 @@ func main() {
 		ReceivedQueries:     receivedQueries,
 	}
 	writeHandler := timeHandler(httpRequestDuration, "write", api.Write(client, elector, &promMetrics))
-	http.Handle("/write", writeHandler)
-	http.Handle("/read", timeHandler(httpRequestDuration, "read", api.Read(client, &promMetrics)))
-	http.Handle("/query", queryHandler)
-	http.Handle("/query_range", queryRangeHandler)
-	http.Handle("/api/v1/query", queryHandler)
-	http.Handle("/api/v1/query_range", queryRangeHandler)
-	http.Handle("/healthz", api.Health(client))
+	router.Post("/write", writeHandler)
+
+	readHandler := timeHandler(httpRequestDuration, "read", api.Read(client, &promMetrics))
+	router.Get("/read", readHandler)
+	router.Post("/read", readHandler)
+
+	queryable := client.GetQueryable()
+	queryEngine := query.NewEngine(log.GetLogger(), time.Minute)
+	queryHandler := timeHandler(httpRequestDuration, "query", api.Query(queryEngine, queryable))
+	router.Get("/query", queryHandler)
+	router.Post("/query", queryHandler)
+	router.Get("/api/v1/query", queryHandler)
+	router.Post("/api/v1/query", queryHandler)
+
+	queryRangeHandler := timeHandler(httpRequestDuration, "query_range", api.QueryRange(queryEngine, queryable))
+	router.Get("/query_range", queryRangeHandler)
+	router.Post("/query_range", queryRangeHandler)
+	router.Get("/api/v1/query_range", queryRangeHandler)
+	router.Post("/api/v1/query_range", queryRangeHandler)
+
+	labelsHandler := timeHandler(httpRequestDuration, "labels", api.Labels(queryable))
+	router.Get("/labels", labelsHandler)
+	router.Post("/labels", labelsHandler)
+	router.Get("/api/v1/labels", labelsHandler)
+	router.Post("/api/v1/labels", labelsHandler)
+
+	labelValuesHandler := timeHandler(httpRequestDuration, "label/:name/values", api.LabelValues(queryable))
+	router.Get("/label/:name/values", labelValuesHandler)
+	router.Get("/api/v1/label/:name/values", labelValuesHandler)
+
```

Comment:
I think we are missing a /api/v1/series endpoint https://prometheus.io/docs/prometheus/latest/querying/api/#finding-series-by-label-matchers

This can be in a separate PR